### ISSUE TITLE
[FEATURE REQ] ADLS: Missing API - PagedFlux onErrorMap function to return a PagedFlux

### ISSUE THREAD

**Issue by gapra-msft** [Azure/azure-sdk-for-java#6750]
**Is your feature request related to a problem? Please describe.**
Related to https://github.com/Azure/azure-sdk-for-java/issues/5825, there is no way to use the onErrorMap function to go from a PagedFlux to a PagedFlux.
PagedFlux extends Flux so the onErrorMap operation returns a Flux.

**Describe the solution you'd like**
To make the namespace for a user consistent, we need to be able to return a DataLakeStorageException.

Storage needs a way to map BlobStorageException errors to DataLakeStorageException errors for ADLS and this is easily achieved by the Reactor .onErrorMap function.

**Additional context**
A code sample of the desired functionality is here
```
            return blobServiceAsyncClient.listBlobContainers(Transforms.toListBlobContainersOptions(options))
                .onErrorMap(ex -> DataLakeImplUtils.transformBlobStorageException(ex))
                .mapPage(Transforms::toFileSystemItem);
```

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [X] Description Added
- [X] Expected solution specified

**Comment by alzimmermsft** [comment]
Adding @anuchandy given his current work may resolve this issue.

**Comment by anuchandy** [comment]
PR that add support for generic decoration of PagedFlux (treat errorMap as one of the decoration): https://github.com/Azure/azure-sdk-for-java/pull/6465

**Comment by anuchandy** [comment]
the implementation based on the above PR looks like:

```java
PagedFlux<FileSystemItem> pagedFlux = blobServiceAsyncClient
                .listBlobContainers(Transforms.toListBlobContainersOptions(options))
                .mapPage(Transforms::toFileSystemItem);

return new PagedFlux<>(() ->
            (Function<String, Flux<PagedResponse< FileSystemItem>>>) ct -> 
               pagedFlux
                 .byPage()
                 .onErrorMap(ex -> DataLakeImplUtils.transformBlobStorageException(ex)));
```

**Comment by gapra-msft** [comment]
@anuchandy Any update on the status of the PR?

**Comment by alzimmermsft** [comment]
This should be resolved by #6465 and will be available in an upcoming release of Azure Core.
